lib/SPIRV/SPIRVReader.h

```diff
@@ -166,6 +168,9 @@ class SPIRVToLLVM {
   typedef DenseMap<SPIRVFunction *, Function *> SPIRVToLLVMFunctionMap;
   typedef DenseMap<GlobalVariable *, SPIRVBuiltinVariableKind> BuiltinVarMap;
   typedef std::unordered_map<SPIRVId, MDNode *> SPIRVToLLVMMDAliasInstMap;
+  typedef DenseMap<SPIRVValue *, std::tuple<PHINode *, SPIRVBasicBlock *, size_t>>
+  SPIRVToLLVMPHINodeArgsMap;
+  typedef DenseSet<SPIRVValue *> SPIRVToLLVMPHINodeArgsSet;
```

Comment:
You should be able to detect presence of a value with `map` as quickly as in within `set`, i.e. the `set` seems to be an extra unneeded thing here